services/address_book_service.proto

```diff
@@ -29,37 +27,127 @@ import "transaction_response.proto";
 import "transaction.proto";
 
 /**
- * Transactions for the AddressBook Service, those HAPI APIs facilitate changes to the nodes used across the Hedera network.
- * All those transactions needs to be signed by the Hedera Council. Steps needed below:
- * 1. The node operator creates and signs the transaction with their key (the key on the node operator account)
- * 2. The node operator hands this transaction to Alex, who then gives it to the council to sign
- * 3. When signed and submitted, the server will verify that account 2 keys have signed, and the keys on the operator account have signed.
- * Hedera council should have ability to make all edits in addition to add/remove nodes
+ * The Address Book service provides the ability for Hedera network node
+ * administrators to add and remove consensus nodes. The addition or removal
+ * of a consensus node requires governing council approval, but each node
+ * operator may update their own operational attributes without additional
+ * approval, reducing overhead for routine operations.
+ *
+ * Most operations are `privileged operations` and require governing council
+ * approval.
+ *
+ * ### For a node creation transaction.
+ * - The node operator SHALL create a `createNode` transaction.
+ *    - The node operator SHALL sign this transaction with the "admin key" of
+ *      the account to be assigned as the "node account".
+ *    - The node operator MUST deliver the signed transaction to the Hedera
+ *      council representative.
+ *    - The Hedera council representative SHALL arrange for council members to
+ *      review and sign the transaction.
+ *    - Once sufficient council members have signed the transaction, the
+ *      Hedera council representative SHALL submit the transaction to the
+ *      network.
+ * - Upon receipt of a valid and signed node creation transaction the network
+ *   software SHALL
+ *    - Validate the threshold signature for the Hedera governing council
+ *    - Validate the signature of the "admin key" for the account to be assigned
+ *      as the "node account".
+ *    - Create the new node in a pending state.
+ *    - When executing the next `freeze` transaction with `freeze_type` set to
+ *      `PREPARE_UPGRADE`, update network configuration and transition the
```

Comment:
When executing the next `freeze` transaction with `freeze_type` set to`PREPARE_UPGRADE`, config.txt, .pem file per each node, and a marker file will be generated.

Reply:
I prefer not to include internal functional details (when _completely_ invisible to external operations) in specification.
Stating what observable transaction causes a state change is helpful to an implementor or client; but including files written to local disk seems an unnecessary detail (another implementation may not do the same thing, it's not observable, and clients shouldn't care).